numpy/fft/_pocketfft.c

```diff
@@ -321,7 +324,29 @@ NOINLINE static void pass2b (size_t ido, size_t l1, const cmplx * restrict cc,
     for (size_t k=0; k<l1; ++k)
       {
       PMC (CH(0,k,0),CH(0,k,1),CC(0,0,k),CC(0,1,k))
-      for (size_t i=1; i<ido; ++i)
+      size_t i=1;
+#if NPY_SIMD_F64
+      size_t step=NPY_SIMD_WIDTH/sizeof(double);
+      for (; i<=ido-step; i+=step)
+        {
+        npyv_f64x2 c, d, tmp, twd;
+        c = npyv_load_deinterleave_f64x2((double*)&cc[(i)+ido*((0)+cdim*(k))]);
```

Comment:
if `cc` is a lookup-table then that will definitely crash on avx2 & avx512f,
due to load more than 2 scalar

Reply:
I'm not sure what you mean by that, but cc is `cmplx[]`. Defined at line 277, `cmplx` is a struct composed of two f64 numbers, `re` and `im`, so sizeof(cmplx) is 16, or 128 bits. In every iteration, the number of cmplx being loaded is `NPY_SIMD_WIDTH/sizeof(double)`, given real and imaginary are stored into two registers.

Reply:
`SSE/NEON/VSX`(128-bit) `npyv_load_deinterleave_f64x2` load 4 f64 from contiguous ptr then deinterleave the payload (2 real and 2 imaginary), and the same goes for `AVX2` and `AVX512` but with x2/x4 payload.

Now I got `Fatal Python error: Segmentation fault` when I tried to run `numpy/fft/tests/test_pocketfft.py` and according to GDB I can tell the segfault caused due to access beyond the pointer boundary.

It seems to me you step by two-dimension over two complex scalars which allow a max 4xf64 for each iteration,
from a quick view, I guess this two-loop needs to be reversed or restructured from the root `pass_all()` in order to
allow a wider SIMD width.

As a workaround please add `#define NPY_SIMD_FORCE_128` before including `simd/simd.h` or at the top of
the source `_pocketfft.c` to ignore `AVX2&AVX512`, and leave a note for it.

Reply:
Misuse of subtraction and comparison betwen unsigned int caused this segmentation fault. This happens when ido is less than step. New commit should do the fix.